src/args.jl

```diff
@@ -2245,19 +2247,14 @@ function _update_series_attributes!(plotattributes::AKW, plt::Plot, sp::Subplot)
 end
 
 function _series_index(plotattributes, sp)
-    idx = 0
-    for series in series_list(sp)
-        if series[:primary]
-            idx += 1
-        end
-        if series == plotattributes
-            return idx
-        end
-    end
-    if get(plotattributes, :primary, true)
-        idx += 1
+    if haskey(plotattributes, :series_index)
+        return plotattributes[:series_index]::Int
+    elseif get(plotattributes, :primary, true)
+        return sp[:max_series_index] =
+            plotattributes[:series_index] = sp[:max_series_index]::Int + 1
```

Comment:
Do we actually need `max_series_index` computing it should be as fast, if I'm not mistaken.
```suggestion
        return plotattributes[:series_index] = length(sp.series_list)::Int + 1
```